packages/app/src/systems/Transaction/utils/tx.ts

```diff
@@ -305,18 +311,40 @@ const mergeAssets = (op1: Operation, op2: Operation) => {
 export function addOperation(operations: Operation[], toAdd: Operation) {
   const ops = operations
     .map((op) => {
+      // if it's not same operation, don't change. we just wanna stackle the same operation
       if (!isSameOperation(op, toAdd)) return null;
-      // if it's not adding any assets, just return the original operation
-      if (!toAdd.assetsSent?.length) return op;
-      // if the original operation doesn't have any assets, just return the toAdd assets
-      if (!op.assetsSent?.length) {
-        return { ...op, assetsSent: toAdd.assetsSent };
+
+      let newOp = { ...op };
+
+      // if it's adding new assets
+      if (toAdd.assetsSent?.length) {
+        // if prev op had assets, merge them. Otherwise just add the new assets
+        newOp = {
+          ...newOp,
+          assetsSent: op.assetsSent?.length
+            ? mergeAssets(op, toAdd)
+            : toAdd.assetsSent,
+        };
+      }
+
+      // if it's adding new calls,
+      if (toAdd.calls?.length) {
+        /*  
+          for calls we don't stack as grouping is not desired.
+          we wanna show all calls in the same operation
+          with each respective assets, amounts, functions, arguments.
+        */
+        newOp = {
+          ...newOp,
+          calls: [...(op.calls || []), ...(toAdd.calls || [])],
+        };
       }
-      // if both have assets, merge them
-      return { ...op, assetsSent: mergeAssets(op, toAdd) };
+
+      return newOp;
```

Comment:
Can we move this logic to another place? If this is related to ABIs, we could add a check if abis are available if not we can skip this.

Reply:
this is not related to ABI.

this sets the way we do/don't stack operations. example
- if 1 contract does 2 transfer to same wallet, we group them and show only 1 transfer of sum value.
- if 1 contract does 2 calls to same function, we don't group them, and show one by one with its respective inputs